packages/dds/tree/src/test/shared-tree/treeNodeApi.spec.ts

```diff
@@ -56,59 +56,480 @@ describe("treeApi", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return { value: view.root.content };
+				});
+				assert.equal(view.root.content, 43);
+				assert.equal(runTransactionResult.success, true);
+				assert.equal(runTransactionResult.value, 43);
+			});
+
+			it("rollback + user defined value", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return { rollback: true, value: view.root.content };
+				});
+				assert.equal(runTransactionResult.success, false);
+				// Note that this is the value that was set before the transaction was rolled back.
+				assert.equal(runTransactionResult.value, 43);
+				// After the transaction is rolled back, the value is reverted to the original value.
+				assert.equal(view.root.content, 42);
+			});
+
+			it("success + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(view.root.content, 43);
+				assert.equal(runTransactionResult.success, true);
+			});
+
+			it("rollback + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						rollback: true,
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(view.root.content, 42);
+				assert.equal(runTransactionResult.success, false);
+			});
+
+			it("success + user defined value + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						value: view.root.content,
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(view.root.content, 43);
+				assert.equal(runTransactionResult.success, true);
+				assert.equal(runTransactionResult.value, 43);
+			});
+
+			it("rollback + user defined value + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						rollback: true,
+						value: view.root.content,
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(runTransactionResult.success, false);
+				// Note that this is the value that was set before the transaction was rolled back.
+				assert.equal(view.root.content, 42);
+				// After the transaction is rolled back, the value is reverted to the original value.
+				assert.equal(runTransactionResult.value, 43);
+			});
+
+			it("void return weirdness", () => {
```

Comment:
I don't think this is testing anything that is unique to `void` or even to this API. Unless I'm misunderstanding something, [this is just how TS works in general](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgJLIN4FgBQz-IBucANgK4QBcyIZAtgEbQDcuAvrrjGSAmMAHsQyGAAoYIaqICUyALwA+NNOqEBwACaYOOTjjEz5S7HgIB6M8gCiADzBQ4yAA5QBT6GACeuAsigQwMihhDCJSCmoAZgAaMPIIACZqABZkNlYcNmkMvVwLNGEAcwgQaFJYgBUAZWQNAQgAZxAAcjBnf0IStrgNDWB+IVJ2tw9gRpFXOmQmUELkIXmGACsIPgaAOjzLCoALFCEST2R+OhR+2vqG453gK9vkAHc94U8BMmb-WtWSOChZ5BIAgQQ2IfzgDBI4zAOzgbQAVHDXmRkDDOsgGvYhIVDgBaLzuLRIqANCAkGDIUDXFAab6-WGCEAIyh6BBCDEU6joOSYHwEYjxKLRXn4fkUJLIZLsPT5Xa3WILaH7RVQFFwEAaWL3GCgKgstltYAJeQ80wi8JUZAxYVxCjJcWSzI5HCskDs4CRTnGw3MIA). So I'd just remove it.

Reply:
I put it there for review purposes. Agreed that it doesn't add any value. Will remove it.

Reply:
removed it.